funnel/views/participant.py

```diff
@@ -130,190 +152,181 @@ class FunnelProjectParticipantView(ProjectParticipantView):
-@lastuser.requires_login
-@load_models(
-    (Profile, {'name': 'profile'}, 'g.profile'),
-    ((Project, ProjectRedirect), {'name': 'project', 'profile': 'profile'}, 'project'),
-    (Participant, {'id': 'participant_id'}, 'participant'),
-    permission='checkin_event',
-)
-def participant_badge(profile, project, participant):
-    return render_template(
-        'badge.html.jinja2', badges=participant_badge_data([participant], project)
-    )
-
-
-@app.route('/<profile>/<project>/participant/<participant_id>/label_badge')
-@funnelapp.route('/<project>/participant/<participant_id>/badge', subdomain='<profile>')
-@lastuser.requires_login
-@load_models(
-    (Profile, {'name': 'profile'}, 'g.profile'),
-    ((Project, ProjectRedirect), {'name': 'project', 'profile': 'profile'}, 'project'),
-    (Participant, {'id': 'participant_id'}, 'participant'),
-    permission='checkin_event',
-)
-def participant_label_badge(profile, project, participant):
-    return render_template(
-        'label_badge.html.jinja2', badges=participant_badge_data([participant], project)
-    )
+@route('/<profile>/<project>/participant/<suuid>')
+class ParticipantView(UrlForView, ModelView):
+    __decorators__ = [legacy_redirect, lastuser.requires_login]
 
+    model = Participant
+    route_model_map = {
+        'profile': 'project.profile.name',
+        'project': 'project.name',
+        'suuid': 'suuid',
+    }
 
-@app.route('/<profile>/<project>/event/<name>/participants/checkin', methods=['POST'])
-@funnelapp.route(
-    '/<project>/event/<name>/participants/checkin',
-    methods=['POST'],
-    subdomain='<profile>',
-)
-@lastuser.requires_login
-@load_models(
-    (Profile, {'name': 'profile'}, 'g.profile'),
-    ((Project, ProjectRedirect), {'name': 'project', 'profile': 'profile'}, 'project'),
-    (Event, {'name': 'name', 'project': 'project'}, 'event'),
-    permission='checkin_event',
-)
-def event_checkin(profile, project, event):
-    form = forms.Form()
-    if form.validate_on_submit():
-        checked_in = getbool(request.form.get('checkin'))
-        participant_ids = request.form.getlist('pid')
-        for participant_id in participant_ids:
-            attendee = Attendee.get(event, participant_id)
-            attendee.checked_in = checked_in
-        db.session.commit()
-        if request.is_xhr:
-            return jsonify(
-                status=True, participant_ids=participant_ids, checked_in=checked_in
+    def loader(self, profile, project, suuid):
+        participant = (
+            self.model.query.join(Project, Profile)
+            .filter(
+                Profile.name == profile,
+                Project.name == project,
+                Participant.suuid == suuid,
             )
-    return redirect(event.url_for('view'), code=303)
+            .first_or_404()
+        )
+        return participant
 
+    def after_loader(self):
+        g.profile = self.obj.project.profile
+        super(ParticipantView, self).after_loader()
```

Comment:
This should be a `return super...` as `after_loader` can optionally return a response.